src/libraries/System.Private.CoreLib/src/System/SpanHelpers.cs

```diff
@@ -404,5 +407,159 @@ public static unsafe void ClearWithReferences(ref IntPtr ip, nuint pointerSizeLe
+                last = ref Unsafe.Add(ref Unsafe.Add(ref first, numBytes), -Vector256<byte>.Count);
+                do
+                {
+                    Vector256<byte> tempFirst = Unsafe.As<byte, Vector256<byte>>(ref first);
+                    Vector256<byte> tempLast = Unsafe.As<byte, Vector256<byte>>(ref last);
+                    tempFirst = Avx2.Shuffle(tempFirst, ReverseMask);
+                    tempFirst = Avx2.Permute2x128(tempFirst, tempFirst, 1);
+                    tempLast = Avx2.Shuffle(tempLast, ReverseMask);
+                    tempLast = Avx2.Permute2x128(tempLast, tempLast, 1);
+                    Unsafe.As<byte, Vector256<byte>>(ref first) = tempLast;
+                    Unsafe.As<byte, Vector256<byte>>(ref last) = tempFirst;
+                    first = ref Unsafe.Add(ref first, Vector256<byte>.Count);
+                    last = ref Unsafe.Add(ref last, -Vector256<byte>.Count);
+                    numBytesWritten += Vector256<byte>.Count * 2;
+                } while (numBytes - numBytesWritten >= Vector256<byte>.Count * 2);
+            }
+            else if (Ssse3.IsSupported && Vector128<byte>.Count * 2 <= numBytes)
+            {
+                Vector128<byte> ReverseMask = Vector128.Create((byte)8, 9, 10, 11, 12, 13, 14, 15, 0, 1, 2, 3, 4, 5, 6, 7);
+                last = ref Unsafe.Add(ref Unsafe.Add(ref first, numBytes), -Vector128<byte>.Count);
+                do
+                {
+                    Vector128<byte> tempFirst = Unsafe.As<byte, Vector128<byte>>(ref first);
+                    Vector128<byte> tempLast = Unsafe.As<byte, Vector128<byte>>(ref last);
+                    tempFirst = Ssse3.Shuffle(tempFirst, ReverseMask);
+                    tempLast = Ssse3.Shuffle(tempLast, ReverseMask);
+                    Unsafe.As<byte, Vector128<byte>>(ref first) = tempLast;
+                    Unsafe.As<byte, Vector128<byte>>(ref last) = tempFirst;
+                    first = ref Unsafe.Add(ref first, Vector128<byte>.Count);
+                    last = ref Unsafe.Add(ref last, -Vector128<byte>.Count);
+                    numBytesWritten += Vector128<byte>.Count * 2;
+                } while ((numBytes - numBytesWritten) >= Vector128<byte>.Count * 2);
+            }
+            last = ref Unsafe.Add(ref first, (numBytes - numBytesWritten) - sizeof(long));
+            ref long firstLong = ref Unsafe.As<byte, long>(ref first);
+            ref long lastLong = ref Unsafe.As<byte, long>(ref last);
+            while (numBytes - numBytesWritten > sizeof(long))
+            {
+                (lastLong, firstLong) = (firstLong, lastLong);
+                firstLong = ref Unsafe.Add(ref firstLong, 1);
+                lastLong = ref Unsafe.Add(ref lastLong, -1);
+                numBytesWritten += sizeof(long) * 2;
+            }
+        }
+
+        public static void Reverse<T>(ref T elements, nuint length)
```

Comment:
If anything in this PR were to be AggressiveInlining, I'd have expected this to be it.  Regardless of the T, this should always compile down to a single call to some other helper.

Reply:
I see, I misunderstood your previous comment. I removed the redundant `AggressiveInlining` you mentioned above and added the attribute to this method. I double checked that the inlining is working as expected with the benchmark, which looks like is inlining as expected:

<details>

<summary>Microbenchmark Disassembly</summary>

## .NET 7.0.0 (42.42.42.42424), X64 RyuJIT
```assembly
; System.Memory.Span`1[[System.Int32, System.Private.CoreLib]].Reverse()
       push      rsi
       sub       rsp,30
       xor       eax,eax
       mov       [rsp+20],rax
       mov       rcx,[rcx+8]
       test      rcx,rcx
       je        short M00_L01
       lea       rax,[rcx+10]
       mov       esi,[rcx+8]
M00_L00:
       mov       [rsp+20],rax
       mov       [rsp+28],esi
       lea       rcx,[rsp+20]
       call      System.MemoryExtensions.Reverse[[System.Int32, System.Private.CoreLib]](System.Span`1<Int32>)
       nop
       add       rsp,30
       pop       rsi
       ret
M00_L01:
       xor       eax,eax
       xor       esi,esi
       jmp       short M00_L00
; Total bytes of code 60
```
```assembly
; System.MemoryExtensions.Reverse[[System.Int32, System.Private.CoreLib]](System.Span`1<Int32>)
       sub       rsp,28
       mov       rdx,[rcx]
       mov       ecx,[rcx+8]
       cmp       ecx,1
       jg        short M01_L00
       add       rsp,28
       ret
M01_L00:
       movsxd    rax,ecx
       mov       rcx,rdx
       mov       rdx,rax
       call      System.SpanHelpers.ReverseInt32Ref(Int32 ByRef, UIntPtr)
       nop
       add       rsp,28
       ret
; Total bytes of code 40
```
```assembly
; System.SpanHelpers.ReverseInt32Ref(Int32 ByRef, UIntPtr)
       vzeroupper
       shl       edx,2
       xor       eax,eax
       cmp       edx,40
       jl        short M02_L01
       vmovupd   ymm0,[7FFD63E2C3E0]
       movsxd    r8,edx
       lea       r8,[rcx+r8+0FFE0]
M02_L00:
       vmovupd   ymm1,[rcx]
       vmovupd   ymm2,[r8]
       vpshufb   ymm1,ymm1,ymm0
       vperm2i128 ymm1,ymm1,ymm1,1
       vpshufb   ymm2,ymm2,ymm0
       vperm2i128 ymm2,ymm2,ymm2,1
       vmovupd   [rcx],ymm2
       vmovupd   [r8],ymm1
       add       rcx,20
       add       r8,0FFFFFFFFFFFFFFE0
       add       eax,40
       mov       r9d,edx
       sub       r9d,eax
       cmp       r9d,40
       jge       short M02_L00
       jmp       short M02_L03
M02_L01:
       cmp       edx,20
       jl        short M02_L03
       vmovupd   xmm0,[7FFD63E2C3E0]
       movsxd    r8,edx
       lea       r8,[rcx+r8+0FFF0]
M02_L02:
       vmovupd   xmm1,[rcx]
       vmovupd   xmm2,[r8]
       vpshufb   xmm1,xmm1,xmm0
       vpshufb   xmm2,xmm2,xmm0
       vmovupd   [rcx],xmm2
       vmovupd   [r8],xmm1
       add       rcx,10
       add       r8,0FFFFFFFFFFFFFFF0
       add       eax,20
       mov       r9d,edx
       sub       r9d,eax
       cmp       r9d,20
       jge       short M02_L02
M02_L03:
       mov       r9d,edx
       sub       r9d,eax
       lea       r8d,[r9+0FFFC]
       movsxd    r8,r8d
       add       r8,rcx
       cmp       r9d,4
       jle       short M02_L05
M02_L04:
       mov       r9d,[rcx]
       mov       r10d,[r8]
       mov       [r8],r9d
       mov       [rcx],r10d
       add       rcx,4
       add       r8,0FFFFFFFFFFFFFFFC
       add       eax,8
       mov       r9d,edx
       sub       r9d,eax
       cmp       r9d,4
       jg        short M02_L04
M02_L05:
       vzeroupper
       ret
; Total bytes of code 227
```

## .NET 7.0.0 (42.42.42.42424), X64 RyuJIT
```assembly
; System.Memory.Span`1[[System.Int32, System.Private.CoreLib]].ReverseArray()
       mov       rcx,[rcx+10]
       jmp       near ptr System.Array.Reverse[[System.Int32, System.Private.CoreLib]](Int32[])
; Total bytes of code 9
```
```assembly
; System.Array.Reverse[[System.Int32, System.Private.CoreLib]](Int32[])
       sub       rsp,28
       test      rcx,rcx
       je        short M01_L00
       lea       rdx,[rcx+10]
       mov       ecx,[rcx+8]
       mov       eax,ecx
       mov       rcx,rdx
       mov       rdx,rax
       add       rsp,28
       jmp       near ptr System.SpanHelpers.ReverseInt32Ref(Int32 ByRef, UIntPtr)
M01_L00:
       mov       ecx,2
       call      System.ThrowHelper.ThrowArgumentNullException(System.ExceptionArgument)
       int       3
; Total bytes of code 44
```
```assembly
; System.SpanHelpers.ReverseInt32Ref(Int32 ByRef, UIntPtr)
       vzeroupper
       shl       edx,2
       xor       eax,eax
       cmp       edx,40
       jl        short M02_L01
       vmovupd   ymm0,[7FFD63E2C580]
       movsxd    r8,edx
       lea       r8,[rcx+r8+0FFE0]
M02_L00:
       vmovupd   ymm1,[rcx]
       vmovupd   ymm2,[r8]
       vpshufb   ymm1,ymm1,ymm0
       vperm2i128 ymm1,ymm1,ymm1,1
       vpshufb   ymm2,ymm2,ymm0
       vperm2i128 ymm2,ymm2,ymm2,1
       vmovupd   [rcx],ymm2
       vmovupd   [r8],ymm1
       add       rcx,20
       add       r8,0FFFFFFFFFFFFFFE0
       add       eax,40
       mov       r9d,edx
       sub       r9d,eax
       cmp       r9d,40
       jge       short M02_L00
       jmp       short M02_L03
M02_L01:
       cmp       edx,20
       jl        short M02_L03
       vmovupd   xmm0,[7FFD63E2C580]
       movsxd    r8,edx
       lea       r8,[rcx+r8+0FFF0]
M02_L02:
       vmovupd   xmm1,[rcx]
       vmovupd   xmm2,[r8]
       vpshufb   xmm1,xmm1,xmm0
       vpshufb   xmm2,xmm2,xmm0
       vmovupd   [rcx],xmm2
       vmovupd   [r8],xmm1
       add       rcx,10
       add       r8,0FFFFFFFFFFFFFFF0
       add       eax,20
       mov       r9d,edx
       sub       r9d,eax
       cmp       r9d,20
       jge       short M02_L02
M02_L03:
       mov       r9d,edx
       sub       r9d,eax
       lea       r8d,[r9+0FFFC]
       movsxd    r8,r8d
       add       r8,rcx
       cmp       r9d,4
       jle       short M02_L05
M02_L04:
       mov       r9d,[rcx]
       mov       r10d,[r8]
       mov       [r8],r9d
       mov       [rcx],r10d
       add       rcx,4
       add       r8,0FFFFFFFFFFFFFFFC
       add       eax,8
       mov       r9d,edx
       sub       r9d,eax
       cmp       r9d,4
       jg        short M02_L04
M02_L05:
       vzeroupper
       ret
; Total bytes of code 227
```
**Method was not JITted yet.**
System.ThrowHelper.ThrowArgumentNullException(System.ExceptionArgument)

</details>

Let me know if there are any other changes I can make.